lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -806,11 +837,17 @@ namespace {
           if (caseItem.getGuardExpr())
             continue;
 
-          auto projection = projectPattern(Ctx, caseItem.getPattern());
-
           // Space is trivially covered with a default clause.
           if (caseItem.isDefault())
             return;
+
+          auto projection = projectPattern(Ctx, caseItem.getPattern());
+          if (projection.isUseful() && projection.isSubspace(Space(spaces))) {
+            Ctx.Diags
+                .diagnose(caseItem.getStartLoc(),
```

Comment:
`git clang-format` did some fun things to this file - it thinks `PAIRCASE`'s that fall through are initializer clauses.